pip/req/req_install.py

```diff
@@ -231,27 +237,23 @@ def build_location(self, build_dir):
         # FIXME: Is there a better place to create the build_dir? (hg and bzr
         # need this)
         if not os.path.exists(build_dir):
+            logger.debug('Creating directory %s', build_dir)
             _make_build_dir(build_dir)
         return os.path.join(build_dir, name)
 
     def correct_build_location(self):
         """If the build location was a temporary directory, this will move it
-        to a new more permanent location"""
+        to a new more permanent location
+        Move self._temp_build_dir to self._ideal_build_dir/self.name
+        """
         if self.source_dir is not None:
             return
         assert self.req is not None
         assert self._temp_build_dir
+        assert self._ideal_build_dir
```

Comment:
This is more of a meta-concern than something for this PR:

Will pip ever be run with with `-O`? I think someone could do: `python -O -m pip install ...` and then these asserts would be removed. Would it be better to change these (in a future PR) to something that won't be skipped?